packages/scaffold/src/views/w3m-wallet-send-view/index.ts

```diff
@@ -0,0 +1,106 @@
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import styles from './styles.js'
+import { CoreHelperUtil, RouterController, SendController } from '@web3modal/core'
+import { state } from 'lit/decorators.js'
+
+@customElement('w3m-wallet-send-view')
+export class W3mWalletSendView extends LitElement {
+  public static override styles = styles
+
+  // -- Members ------------------------------------------- //
+  private unsubscribe: (() => void)[] = []
+
+  // -- State & Properties -------------------------------- //
+  @state() private token = SendController.state.token
+
+  @state() private sendTokenAmount = SendController.state.sendTokenAmount
+
+  @state() private receiverAddress = SendController.state.receiverAddress
+
+  @state() private message:
+    | 'Preview Send'
+    | 'Select Token'
+    | 'Add Address'
+    | 'Add Amount'
+    | 'Insufficient Funds'
+    | 'Invalid Address' = 'Preview Send'
+
+  public constructor() {
+    super()
+    this.unsubscribe.push(
+      ...[
+        SendController.subscribe(val => {
+          this.token = val.token
+          this.sendTokenAmount = val.sendTokenAmount
+          this.receiverAddress = val.receiverAddress
+        })
+      ]
+    )
+  }
+
+  public override disconnectedCallback() {
+    this.unsubscribe.forEach(unsubscribe => unsubscribe())
+  }
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    if (this.token) {
+      if (this.sendTokenAmount) {
+        if (this.sendTokenAmount > Number(this.token.quantity.numeric)) {
+          this.message = 'Insufficient Funds'
+        } else if (this.receiverAddress) {
+          this.message = CoreHelperUtil.isAddress(this.receiverAddress)
+            ? 'Preview Send'
+            : 'Invalid Address'
+        } else {
+          this.message = 'Add Address'
+        }
+      } else {
+        this.message = 'Add Amount'
+      }
+    } else {
+      this.message = 'Select Token'
+    }
```

Comment:
```suggestion
    if(!this.token) {
      this.message = 'Select Token'
    }

    if(!this.sendTokenAmount) {
      this.message = 'Add Amount'
    }

    if(this.sendTokenAmount > Number(this.token.quantity.numeric)) {
      this.message = 'Insufficient Funds'
    }

    if(!this.receiverAddress) {
      this.message = 'Add Address'
    }

    if(!CoreHelperUtil.isAddress(this.receiverAddress)) {
      this.message = 'Invalid Address'
    }

    this.message = 'Preview Send'
```

Reply:
I think in this case using the ifs separately is much clearer. I'd in fact add an util method like `getMessage` or similar